File: back/src/main/java/com/clinic/entities/Order.java
package com.clinic.entities;

import com.fasterxml.jackson.annotation.*;
import lombok.*;
import org.springframework.web.bind.annotation.GetMapping;

import javax.persistence.*;
import javax.validation.constraints.NotEmpty;
import java.util.List;
import java.util.Set;

@Table(name = "orders")
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private String comment;

    private String confirmation;

    @JsonManagedReference
    @ManyToOne
    @JoinColumn(name = "accompaniment_script")
    private AccompanimentScript accompanimentScript;

    private String genome;

    @JsonManagedReference
    @ManyToOne
    @JoinColumn(name = "id_client")
    private Client client;

    @JsonManagedReference
    @ManyToOne
    @JoinColumn(name = "spec")
    private Specialization specialization;

    @ManyToMany
    @JoinTable(
            name = "orders_modifications",
            joinColumns = @JoinColumn(name = "id_order"),
            inverseJoinColumns = @JoinColumn(name = "id_mod"))
    Set<Modification> modifications;

    @JsonManagedReference
    @OneToMany(mappedBy = "order")
    private List<BodyChange> bodyChanges;

    @JsonManagedReference
    @OneToOne(cascade = CascadeType.ALL, mappedBy = "order")
    private Scenario scenario;

}

File: front/SAUT/src/app/_models/Order.ts
import * as internal from 'assert'
import {Client} from './Client'

export interface Order {
  id: number;
  comment:string;
  confirmation:string;
  accompanimentScript: Accompaniment;
  genome: string;
  client: Client;
  specialization: Spec;
  modifications: Mod[];
  bodyChanges: BodyChange[];
  scenario: Scenario;
}

export interface Spec {
  id: number, 
  name: string
}

export interface Accompaniment{
  id: number;
  scenarios: string;
}

export interface Mod {
  id: number;
  name: string;
  price: number;
  currency: string;
  risk: string;
  chance: number;
  accompaniment: string;
}

export interface BodyChange{
  id:number;
  change: string;
  description: string;
  symptoms: string;
  actions: string;
}

export interface Scenario{
  id: number;
  modificationScenarios: modScen;
}
export interface modScen{
  modification: Mod;
  priority: number;
}

